weaviate/auth.py

```diff
@@ -77,20 +77,54 @@ def __post_init__(self) -> None:
 
 
 @dataclass
-class APIKey:
+class _APIKey:
     """Using the given API key to authenticate with weaviate."""
 
     api_key: str
 
 
-OidcAuth = Union[BearerToken, ClientPassword, ClientCredentials]
-AuthCredentials = Union[OidcAuth, APIKey]
+class Auth:
+    @staticmethod
+    def api_key(api_key: str) -> _APIKey:
+        return _APIKey(api_key)
 
-AuthApiKey = APIKey
+    @staticmethod
+    def client_credentials(
+        client_secret: str, scope: Optional[SCOPES] = None
+    ) -> _ClientCredentials:
+        return _ClientCredentials(client_secret, scope)
+
+    @staticmethod
+    def client_password(
+        username: str, password: str, scope: Optional[SCOPES] = None
+    ) -> _ClientPassword:
+        return _ClientPassword(username=username, password=password, scope=scope)
+
+    @staticmethod
+    def bearer_token(
+        bearer_token: str, expires_in: int = 60, refresh_token: Optional[str] = None
+    ) -> _BearerToken:
+        return _BearerToken(bearer_token)
+
+
+OidcAuth = Union[_BearerToken, _ClientPassword, _ClientCredentials]
+AuthCredentials = Union[OidcAuth, _APIKey]
+
+
+APIKey = _APIKey
+"""@deprecated; use wvc.Auth.api_key() instead."""
+BearerToken = _BearerToken
+"""@deprecated; use wvc.Auth.bearer_token() instead."""
+ClientCredentials = _ClientCredentials
+"""@deprecated; use wvc.Auth.client_credentials() instead."""
+ClientPassword = _ClientPassword
+"""@deprecated; use wvc.Auth.client_password() instead."""
+
+AuthApiKey = _APIKey
 """@deprecated; use APIKey instead."""
-AuthBearerToken = BearerToken
+AuthBearerToken = _BearerToken
 """@deprecated; use BearerToken instead."""
-AuthClientCredentials = ClientCredentials
+AuthClientCredentials = _ClientCredentials
 """@deprecated; use ClientCredentials instead."""
-AuthClientPassword = ClientPassword
+AuthClientPassword = _ClientPassword
 """@deprecated; use ClientPassword instead."""
```

Comment:
Should these deprecated messages point towards `wvc.Auth.` instead?

It's the kind of thing that could be picked up in the docstrings PR but it's also so small that it's easy to miss